src/nvim/message.c

```diff
@@ -1124,6 +1163,24 @@ void set_keep_msg(char_u *s, int attr)
   keep_msg_attr = attr;
 }
 
+void msg_set_ext_kind(const char *msg_kind)
```

Comment:
`msg_ext_set_kind`

More generally, some of these flags seem potentially useful for the legacy message system. It's not clear to me if they should be considered "external only".

Reply:
Well, their names are not public or anything, they could be changed later. (Also it is an useful  ambiguity that `ext` also can mean "extended". If lua will render internal popupmenu from "ext" events, `ext_popupmenu` then means "extended popupmenu")